lib/protocol/src/openctx-protocol.schema.ts

```diff
@@ -62,11 +69,32 @@ export interface Selector {
      */
     contentContains?: string
 }
-export interface ItemsParams {
+/**
+ * The viewer context sent by client to the provider. 
+ */
+export interface Viewer {
+    /**
+     * The list of respositoy remotes from the workspace.
+     * Eg. ["github.com/sourcegraph/openctx"]
+     */
+    repositoryRemotes?: string[]    
+}
+export interface CandidateItemsParams<K extends ItemKindID = ItemKindID> {
+    /**
+     * A search query that is interpreted by providers to find the candidates items.
+     */
+    query?: string
+    kind?: K
+    viewer?: Viewer
+}
+export interface ItemsParams<K extends ItemKindID = ItemKindID>{
     /**
      * A search query that is interpreted by providers to filter the items in the result set.
      */
     query?: string
+    kind?: K
+    viewer?: Viewer
+    candidateItems?: CandidateItem[]
```

Comment:
any chance we are making provider implementors lifes difficult by having such a wide param struct for both candidate and items function calls?

If I remember the original motivation for making it this big is we are thinking ahead to non-@mention cases where query can be a chat query?